db.go

```diff
@@ -795,10 +801,25 @@ func (db *DB) createGeneration(ctx context.Context) (string, error) {
 }
 
 // Sync copies pending data from the WAL to the shadow WAL.
-func (db *DB) Sync(ctx context.Context) (err error) {
-	db.mu.Lock()
-	defer db.mu.Unlock()
+func (db *DB) Sync(ctx context.Context) error {
+	const retryN = 5
+
+	for i := 0; i < retryN; i++ {
+		if err := func() error {
+			db.mu.Lock()
+			defer db.mu.Unlock()
+			return db.sync(ctx)
+		}(); err != nil {
+			db.Logger.Printf("sync error, retrying: %s", err)
+		} else {
+			break
+		}
+	}
```

Comment:
We no longer check the database for changes every second so we are attempting retries instead.